File: kt/Minsk/src/main/kotlin/dev/phytolizer/minsk/SyntaxKind.kt
package dev.phytolizer.minsk

enum class SyntaxKind {
    BadToken,
    EndOfFileToken,

    WhitespaceToken,
    NumberToken,

    PlusToken,
    MinusToken,
    StarToken,
    SlashToken,
    OpenParenthesisToken,
    CloseParenthesisToken,

    BinaryExpression,
    LiteralExpression,
    UnaryExpression,
    ParenthesizedExpression,
}

File: hs/src/Minsk/SyntaxKind.hs
module Minsk.SyntaxKind (SyntaxKind (..)) where

import BasicPrelude

data SyntaxKind
  = -- Special tokens
    BadToken
  | EndOfFileToken
  | WhitespaceToken
  | -- Single-char tokens
    PlusToken
  | MinusToken
  | StarToken
  | SlashToken
  | OpenParenthesisToken
  | CloseParenthesisToken
  | -- Multi-char tokens
    NumberToken
  | -- Expressions
    BinaryExpression
  | LiteralExpression
  | ParenthesizedExpression
  | UnaryExpression
  deriving (Eq, Show)
